symtabAPI/h/Type.h

```diff
@@ -435,16 +435,53 @@ class SYMTAB_EXPORT typeFunction : public Type {
 typeFunction& Type::asFunctionType() { return dynamic_cast<typeFunction&>(*this); }
 
 class SYMTAB_EXPORT typeScalar : public Type {
- private:
-   bool isSigned_;
- public:
-   typeScalar();
-   typeScalar(typeId_t ID, unsigned int size, std::string name = "", bool isSigned = false);
-   typeScalar(unsigned int size, std::string name = "", bool isSigned = false);
-   static typeScalar *create(std::string &name, int size, Symtab *obj = NULL);
-   bool isSigned();
-   bool isCompatible(boost::shared_ptr<Type> x) { return isCompatible(x.get()); };
-   bool isCompatible(Type *otype);
+public:
+  struct properties_t {
+	  // Summary properties
+	  // NB: See DwarfWalker::parseBaseType for how these are computed
+	  bool is_integral;
+	  bool is_floating_point;
+	  bool is_string;
+
+	  // Detailed properties
+	  bool is_address;
```

Comment:
What about some kind of boolean for is_null, would that be useful?

Reply:
I was looking at this, and I think we can add it here. DWARF treats 'void' separately from the base_type attributes handled here, but I think we can lump them together. I was going to chat with @sashanicolas first before adding it.